From 9ac10f0dc60c63dbd21b7ae48cdeb53b41900acd Mon Sep 17 00:00:00 2001
From: JasonHuang <Jason.Huang2@infineon.com>
Date: Tue, 18 Oct 2022 01:06:04 -0500
Subject: [PATCH 153/208] brcmfmac: remove workaround cause the FW can support
 6G channels

Earlier firmware not ready support 6G channels. We add this workaround
to avoid FW crash, when driver want to remove disabled 6G channels. Now,
remove this workaround cause firmware is ready for the 6G channels.

Signed-off-by: JasonHuang <Jason.Huang2@infineon.com>
---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
index 4685fdfa85cc..a9c9639d86dd 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
@@ -8078,9 +8078,6 @@ static int brcmf_construct_chaninfo(struct brcmf_cfg80211_info *cfg,
 					next = &band->channels[j + 1];
 					memcpy(cur, next, sizeof(*cur));
 				}
-			/* To avoid fw crash while delete all channels */
-				if (n_6g == 1)
-					break;
 				n_6g--;
 			} else
 				i++;
-- 
2.17.1

